core/src/main/java/tc/oc/pgm/filters/dynamic/FilterMatchModule.java

```diff
@@ -89,10 +104,29 @@ private static class ListenerSet {
   // Filterables that need a check in the next tick (cleared every tick)
   private final Set<Filterable<?>> dirtySet = new HashSet<>();
 
-  // We have to listen to this event to guarantee that all other modules have been loaded
-  // and that there is no unresolved xml filter references.
   @EventHandler
   public void onMatchLoad(MatchLoadEvent event) {
+
+    // These two have to be on separate passes, in case a reactor wants to register a child filter
+    // for dynamic listening. This could potentially result in a need to register a specific event
+    // listener.
+
+    // First, find and create all necessary reactors
+    this.filterContext
+        .getAllUnchecked(ReactorFactory.class)
+        .forEach(
+            filter -> {
+              final ReactorFactory.Reactor reactor = filter.createReactor(this.match, this);
+              this.activeReactors.put(
+                  (ReactorFactory<? extends ReactorFactory.Reactor>) filter, reactor);
+              System.out.println(
+                  "_________________Activated reactor " + filter + "______________________");
+            });
+    this.filterContext // Check for any factories hidden in filter trees
+        .getAllUnchecked(ParentFilter.class)
+        .forEach(this::findAndCreateReactorFactories);
```

Comment:
Can't this all be replaced with simply doing findAndCreateReactorFactories with all the Filter.class instances, instead of doing ReactorFactory first and ParentFilter second?
Also, do we know for certain that all filters are registered to the context (regardless of if they have an id or not)?

Reply:
Yes, all filters parsed by the filter parsers are added to a context in all cases. If no id is used it just means that you have to iterate over everything like i'm doing here to discover it.